pkg/providers/launchtemplate/launchtemplate.go

```diff
@@ -424,3 +431,26 @@ func (p *Provider) DeleteLaunchTemplates(ctx context.Context, nodeClass *v1beta1
 	}
 	return nil
 }
+
+func (p *Provider) ResolveClusterCIDR(ctx context.Context) error {
+	if p.ClusterCIDR.Load() != nil {
```

Comment:
Does this ever need to be staled out? I don't think this can change, so I don't think so.

Reply:
Nope, cluster CIDR is immutable. The only reason we don't discover it at launch is because it's only required for AL2023. @jonathan-innis and I had a discussion offline about where to resolve this and landed on NodeClass reconciliation.